client/lib/signup/step-actions.js

```diff
@@ -401,7 +401,11 @@ export function createAccount(
 					analytics.ga.recordEvent( 'Signup', 'calypso_user_registration_complete' );
 				}
 
-				const providedDependencies = assign( {}, { username: userData.username }, bearerToken );
+				const username = response.is_signup_sandbox
```

Comment:
Nitpick alert 🚨 

This could probably be rewritten as:

```suggestion
				const username = response.signup_sandbox_username || userData.username;
```

Not a biggie, of course.